frontend/simple/views/components/ProposalBox/ProposalBox.vue

```diff
@@ -0,0 +1,305 @@
+
+export default {
+  name: 'ProposalBox',
+  props: {
+    proposal: {
+      type: [Object],
+      required: true
+    },
+    // Actions:
+    onVoteAgainst: Function,
+    onVoteFor: Function,
+    onCloseProposal: Function
+  },
+  components: {
+    ButtonCountdown,
+    Sign
+  },
+  data () {
+    return {
+      ephemeral: {
+        errorState: null,
+        closeCountdown: {
+          state: null,
+          countdown: null
+        }
+      }
+    }
+  },
+  computed: {
+    buttonText () {
+      const { doWhat, toWhat, fromWhat } = this.proposal
+      if (doWhat === DoInvitation) {
+        return {
+          for: template(L('Invite {toWhat}'), { toWhat }),
+          against: L('Don\'t invite')
+        }
+      } else if (doWhat === DoRemoval) {
+        return {
+          for: template(L('Remove {toWhat}'), { toWhat }),
+          against: template(L('Keep {toWhat}'), { toWhat })
+        }
+      } else {
+        return {
+          for: template(L('Change to {toWhat}'), { toWhat }),
+          against: template(L('Keep {fromWhat}'), { fromWhat })
+        }
+      }
+    },
+    title () {
+      const titleMap = {
+        [DoInvitation]: L('Invite Member'),
+        [DoRemoval]: L('Remove Member'),
+        [DoIncome]: L('Change Mincome'),
+        [DoChangeThreshold]: L('Update Rule: Change Threshold'),
+        [DoApprovalThreshold]: L('Update Rule: Invite Threshold'),
+        [DoRemovalThreshold]: L('Update Rule: Member Removal Threshold')
+      }
+      return titleMap[this.proposal.doWhat]
+    },
+    text () {
+      const { whoProposed, doWhat, toWhat, fromWhat } = this.proposal
+      const proposer = this.isOwnProposal ? L('You') : whoProposed
+      const textMap = {
+        [DoInvitation]: template(
+          L('{proposer} proposed to <strong>invite {toWhat}</strong> to the group'), {
+            proposer, toWhat }
+        ),
+        [DoRemoval]: template(
+          L('{proposer} proposed to <strong>remove {toWhat}</strong> from the group'), {
+            proposer, toWhat }
+        ),
+        [DoIncome]: template(
+          L('{proposer} proposed to change the <strong>mincome from {fromWhat} to {toWhat}</strong>'), {
+            proposer, toWhat, fromWhat }
+        ),
+        [DoChangeThreshold]: template(
+          L('{proposer} proposed to change the <strong>rule changing threshold from {fromWhat} to {toWhat}</strong>'), {
+            proposer, toWhat, fromWhat }
+        ),
+        [DoApprovalThreshold]: template(
+          L('{proposer} proposed to change the <strong>member invite threshold from {fromWhat} to {toWhat}</strong>'), {
+            proposer, toWhat, fromWhat }
+        ),
+        [DoRemovalThreshold]: template(
+          L('{proposer} proposed to change the <strong>member removal threshold from {fromWhat} to {toWhat}</strong>'), {
+            proposer, toWhat, fromWhat }
+        )
+      }
+      return textMap[doWhat]
```

Comment:
We have too many maps. It seems like creating a single map in a separate file is a good idea. Perhaps a `proposals.js` file.